### DISCUSSION TITLE
Partial decompression of remote file

### DISCUSSION THREAD

**Discussion by froody** [Blosc/c-blosc2#609] [Q&A]
I want to do partial reads of a remote file compressed with blosc2, transferring as few bytes as possible. Given a list of (offset,extent) ranges in the decompressed output, I'd like to be able to generate a list of (offset, extent) ranges in the compressed file to fetch and then somehow decompress this data with blosc2. I'd like to do this in as few round-trips as possible, ideally one to fetch the header + bstarts, and one to fetch the ranges of compressed data. I have complete control over the generation of the compressed file, and it's also an option to pre-process them somehow (e.g. extracting the header + bstarts out into a separate file for easy lookup).

Is this at all possible with the public api? Is there some example code I should look at? So far I'm thinking about calling `blosc2_schunk_get_chunk` for each chunk during preprocessing, saving that somewhere, and then using `blosc2_io_cb` to define my own reader to read from the downloaded fragments.

**Comment by FrancescAlted** [comment]
Yes, [lazy chunks](https://github.com/Blosc/c-blosc2/blob/main/include/blosc2.h#L2026) are perfect for that.  Look at [the format](https://github.com/Blosc/c-blosc2/blob/main/README_CHUNK_FORMAT.rst) of the lazychunk to understand how bstarts and [bsizes](https://github.com/Blosc/c-blosc2/blob/main/README_CHUNK_FORMAT.rst#trailer) are stored.
